tools/install-powershell.ps1

```diff
@@ -79,6 +79,68 @@ Function Remove-Destination([string] $Destination) {
+    [string] $HKCURegKeyPath = Get-ItemProperty -Path 'HKCU:\Environment' -Name 'PATH' -ErrorAction SilentlyContinue
+    $InstalledPaths += @([System.Environment]::ExpandEnvironmentVariables(($HKCURegKeyPath)).Split(';'))
+    [string] $HKLMRegKeyPath = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment' -Name 'PATH' -ErrorAction SilentlyContinue
+    $InstalledPaths += @([System.Environment]::ExpandEnvironmentVariables(($HKLMRegKeyPath)).Split(';'))
+    #Remove ending DirectorySeparatorChar in all items of array for comparison purposes
+    $InstalledPaths = $InstalledPaths.ForEach( { $_.Replace('\\$', '') })
+    if ($InstalledPaths -icontains $Path) {
+        return $True
+    }
+    return $False
+}
+
+Function SavePathInRegistryKey([Microsoft.Win32.RegistryKey] $Key, [string] $Path) {
+    [string] $CurrentUnexpandedValue = $Key.GetValue('PATH', '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
+    #Keep current PathValueKind if possible/appropriate
+    try {
+        [Microsoft.Win32.RegistryValueKind]$PathValueKind = $Key.GetValueKind('PATH')
+    } catch {
+        [Microsoft.Win32.RegistryValueKind]$PathValueKind = [Microsoft.Win32.RegistryValueKind]::ExpandString
+    }
+    #Evaluate new path
+    $NewPathValue = [string]::Concat(
+        [regex]::Replace($CurrentUnexpandedValue, ([System.IO.Path]::DirectorySeparatorChar + '$')),
+        [System.IO.Path]::DirectorySeparatorChar, $Path)
+    #Upgrade PathValueKind to [Microsoft.Win32.RegistryValueKind]::ExpandString if appropriate
+    if ($NewPathValue.Contains('%')) { $PathValueKind = [Microsoft.Win32.RegistryValueKind]::ExpandString }
+    $Key.SetValue("PATH", $NewPathValue, $PathValueKind)
+}
+
+Function AddPathToPathWinEnv([string]$Path) {
+    if (-not (IsPathInRegistryPath($Path))) {
+        #Do not save user paths in machine wide registry path
+        #USERPROFILE is not a subpath of APPDATA or LOCALAPPDATA for roaming profiles
+        #APPDATA and LOCALAPPDATA are relocatable ==> test all 3
+        if ((-not ($Path.StartsWith($ENV:USERPROFILE))) -and
+            (-not ($Path.StartsWith($ENV:APPDATA))) -and
+            (-not ($Path.StartsWith($env:LOCALAPPDATA)))) {
+            try {
+                [string]$KeyName = 'SYSTEM\CurrentControlSet\Control\Session Manager\Environment\'
+                SavePathInRegistryKey([Microsoft.Win32.Registry]::LocalMachine.OpenSubKey($KeyName), $Path)
+            } catch {
+                Write-Verbose -Message "Unable to save the new path in the machine wide registry."
+            }
+        }
+    }
+    #If failed to install to machine wide path or path was not appropriate for machine wide path
+    if (-not (IsPathInRegistryPath($Path))) {
+        try {
+            [string]$KeyName = 'Environment'
+            SavePathInRegistryKey([Microsoft.Win32.Registry]::CurrentUser.OpenSubKey($KeyName), $Path)
```

Comment:
```suggestion
[System.Environment]::SetEnvironmentVariable('PATH',$Path,'User')
``````

Reply:
but.... this would reintroduce the problems this PR is trying to avoid!  may as well close the pr then.

Reply:
(no changes to code)